src/mongo/db/pipeline/expression.cpp

```diff
@@ -490,8 +490,15 @@ const char* ExpressionArray::getOpName() const {
 /* ------------------------- ExpressionArrayElemAt -------------------------- */
 
 Value ExpressionArrayElemAt::evaluate(const Document& root) const {
-    const Value array = vpOperand[0]->evaluate(root);
     const Value indexArg = vpOperand[1]->evaluate(root);
+    long long i = indexArg.coerceToLong();
```

Comment:
This coerceToLong with throw an error if it's not a long, and that error message will not have as much context as the error that used to get thrown on line 511. Please move the assert on line 511 up before this, then use getLong instead: https://github.com/mongodb/mongo/blob/d50f8ba3c8f3f9380876361def45f8531247f8aa/src/mongo/db/pipeline/value.h#L203